text/0000-crate-changelogs.md

```diff
@@ -0,0 +1,237 @@
+- Feature Name: crate-changelogs
+- Start Date: 2017-08-27
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add changelog support to `cargo` and [crates.io](https://crates.io/).
+
+# Motivation
+[motivation]: #motivation
+
+Citing [keepachangelog.com](http://keepachangelog.com):
+
+> What is a changelog?
+>
+>  A changelog is a file which contains a curated, chronologically ordered list of notable changes for each version of a project.
+>
+> Why keep a changelog?
+>
+> To make it easier for users and contributors to see precisely what notable changes have been made between each release (or version) of the project.
+>
+> Who needs a changelog?
+>
+> People do. Whether consumers or developers, the end users of software are
+> human beings who care about what's in the software. When the software changes,
+> people want to know why and how.
+
+Encouraging crate authors to keep changelogs and increase its visibility for
+crate users will definitely benefit crates ecosystem.
+
+This topic was brought several times, most notable notions are:
+
+- [Rustaceans: Please Keep a Changelog!](https://blog.dbrgn.ch/2015/12/1/rust-crates-keep-a-changelog/) ([reddit](https://www.reddit.com/r/rust/comments/3v1ndl/))
+- [Cargo issue](https://github.com/rust-lang/cargo/issues/2188)
+- [Recent reddit thread](https://www.reddit.com/r/rust/comments/6vvhjh/)
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+There is two ways to add changelog support to your crate: providing changelog
+file explicitly or linking to an external resource.
+
+## Explicit changelog file
+
+Add `changelog` field to `[package]` section of your `Cargo.toml` with the
+relative path to your changelog file:
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+changelog = "CHNAGELOG.md"
```

Comment:
There's a small typo here. `CHNAGELOG.md` -> `CHANGELOG.md`

Reply:
Thank you for noticing!